### PR TITLE
[SofaMiscForceField] Fix MeshMatrixMass duplicate Data parameters

### PR DESCRIPTION
**Pull request by epernod**
There was normal vector Data:
```
Data< sofa::helper::vector< Real > > d_vertexMass;
Data< sofa::helper::vector< Real > > d_edgeMass;
...
, d_vertexMass( initData(&d_vertexMass, "vertexMass", "Specify a vector giving the mass of each vertex. If unspecified or wrongly set, another mass information is used.") )
, d_edgeMass( initData(&d_edgeMass, "edgeMass", "values of the particles masses on edges") )
```
and topological Data with mechanism to update each other then they are changed or topological changes occurs.
```
/// Values of the particles masses stored on vertices
topology::PointData<helper::vector<MassType> >  d_vertexMass;
/// Values of the particles masses stored on edges
topology::EdgeData<helper::vector<MassType> >   d_edgeMass;
...
, d_vertexMassInfo( initData(&d_vertexMassInfo, "vertexMassInfo", "internal values of the particles masses on vertices, supporting topological changes") )
, d_edgeMassInfo( initData(&d_edgeMassInfo, "edgeMassInfo", "internal values of the particles masses on edges, supporting topological changes") )
```

No need to have 2 sets of Data, topological Data can be used directly as PointData and EdgeData inherite from Data and support topological changes.

-> Remove the normal Data but keep the names vertexMass and edgeMass which are used in code and scenes.

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by epernod** [comment]
[ci-build][with-all-tests]

**Review by alxbilger — APPROVED** [review]

File: modules/SofaMiscForceField/src/SofaMiscForceField/MeshMatrixMass.inl [hunk 1]
```diff
@@ -1230,8 +1223,8 @@ void MeshMatrixMass<DataTypes, MassType>::computeMass()
     clear();
 
     // prepare to store info in the vertex array
-    helper::vector<MassType>& my_vertexMassInfo = *d_vertexMassInfo.beginEdit();
-    helper::vector<MassType>& my_edgeMassInfo = *d_edgeMassInfo.beginEdit();
+    helper::vector<MassType>& my_vertexMassInfo = *d_vertexMass.beginEdit();
```
Comment (fredroy) [inline modules/SofaMiscForceField/src/SofaMiscForceField/MeshMatrixMass.inl:1226]: Use WriteOnlyAccessor() instead of the dangerous beginEdit()

**Review by fredroy — CHANGES_REQUESTED** [review]

File: modules/SofaMiscForceField/src/SofaMiscForceField/MeshMatrixMass.inl (hunk 1 above)
Comment (epernod) [reply modules/SofaMiscForceField/src/SofaMiscForceField/MeshMatrixMass.inl:1226]: this is already done in next PR #2193 
This one is just a Ctrl + shift + H

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

**Pull request merged**